src/menu.scss

```diff
@@ -5,165 +5,303 @@
+      }
+    }
+
+    @include fd-screen(m) {
+      width: 2.25rem; // to account for item right padding of 16px or 1rem so 2.25rem-1rem = 1.25rem
+      height: 1.25rem; // to account for item top padding of 12px or 0.75rem so 2.75rem-0.75rem-0.75rem = 1.25rem
+      margin: 0 0 0 -0.75rem;
+
+      @include fd-rtl() {
+        margin: 0 -0.1rem 0 0; // since before and after icons switch places, their margins also switch
+      }
+    }
+
+    @include fd-screen(l) {
+      width: 2rem; // to account for item right padding of 12px or 0.75rem so 2rem-0.75rem = 1.25rem
+      height: 1.25rem; // to account for item top padding of 6px or 0.375rem so 2rem-0.375rem-0.375rem = 1.25rem
+      margin: 0 0 0 -0.5rem;
+
+      @include fd-rtl() {
+        margin: 0 -0.75rem 0 0; // since before and after icons switch places, their margins also switch
+      }
     }
   }
 
-  &--addon-before {
-    .fd-menu__item {
-      padding-top: $fd-menu-item-padding-y;
-      padding-bottom: $fd-menu-item-padding-y;
-      padding-left: 2px;
-      padding-right: 2px;
+  @mixin icon-after-size-responsive() {
+    @include fd-screen(s) {
+      width: 2.75rem; // to account for item right padding of 16px or 1rem so 2.75rem-1rem = 1.75rem
+      height: 1.25rem; // to account for item top padding of 12px or 0.75rem so 2.75rem-0.75rem-0.75rem = 1.25rem
+      margin: 0 -1rem 0 0;
+
+      @include fd-rtl() {
+        margin: 0 0 0 -1rem; // since before and after icons switch places, their margins also switch
+      }
+    }
+
+    @include fd-screen(m) {
+      width: 2.25rem; // to account for item right padding of 16px or 1rem so 2.25rem-1rem = 1.25rem
+      height: 1.25rem; // to account for item top padding of 12px or 0.75rem so 2.75rem-0.75rem-0.75rem = 1.25rem
+      margin: 0 -1rem 0 0;
+
+      @include fd-rtl() {
+        margin: 0 0 0 -0.75rem; // since before and after icons switch places, their margins also switch
+      }
+    }
+
+    @include fd-screen(l) {
+      width: 2rem; // to account for item right padding of 12px or 0.75rem so 2rem-0.75rem = 1.25rem
+      height: 1.25rem; // to account for item top padding of 6px or 0.375rem so 2rem-0.375rem-0.375rem = 1.25rem
+      margin: 0 -0.75rem 0 0;
+
+      @include fd-rtl() {
+        margin: 0 0 0 -0.5rem; // since before and after icons switch places, their margins also switch
+      }
     }
   }
 
   &__addon-before {
+    @include fd-reset();
     @include fd-flex-center();
 
     @include fd-rtl() {
       float: right;
       clear: right;
     }
 
+    color: $fd-menu-primary-icon-color;
+
+    @include fd-active() {
+      color: $fd-menu-item-color--active;
+    }
+
+    @include fd-selected() {
+      color: $fd-menu-primary-icon-color;
+
+      @include fd-hover() {
+        color: $fd-menu-item-color--active;
+      }
+    }
+
+    flex: 0 0 auto;
+    display: flex;
     background-color: transparent;
     float: left;
```

Comment:
Try to remove floats, it's out of control. There are better alternatives

Reply:
removed